### DISCUSSION TITLE
Merge with duplicate primary key is not working as expected

### DISCUSSION THREAD

**Discussion by serkancoban** [kuzudb/kuzu#3102] [Q&A]
Hi, we expect below query to complete successfully, but it gives error:
LOAD FROM "kuzu.parquet" WITH col1 as c1,col2 as c2, col3 as c3  limit 10 merge (n:User {id:string(id)});
Runtime exception: Found duplicated primary key value 1234, which violates the uniqueness constraint of the primary key column

Merge normally should handle this right?

**Comment by andyfengHKU** [comment]
Hi @serkancoban, you are correct.

This is definitely a bug on our side. I'll take a look at and hopefully fix it in a few days.

**Comment by prrao87** [comment]
I've moved this over to #3103.
